languages/rust/exercises/concept/string-vs-str/.docs/introduction.md

```diff
@@ -0,0 +1,56 @@
+`String` is a potentially-mutable utf8-encoded representation of a sequence of Unicode codepoints. (In Java or C#, this type is called `StringBuilder`.)
```

Comment:
This file is well-written! It is slightly on the longer side though. As per [the spec](https://github.com/exercism/v3-docs/blob/master/anatomy/tracks/concept-exercises.md#file-docsintroductionmd):

> The information provided should give the student just enough context to figure out the solution themselves.

What this means is that this file contain _just enough_ information for the student to figure things out by themselves. There are often aspects of a concept that are very interesting, but that the student does not need to understand in order to solve the exercise. This type of information should be removed from the `introduction.md` file. There is another place to put this information though, which is the concept's `about.md` file (see [the spec](https://github.com/exercism/v3-docs/blob/master/anatomy/tracks/concepts.md#file-aboutmd)). This file is shown once the student finishes the exercise, and can go in far more detail on a concept than the `introduction.md` allows for.

I'll leave it up to you to decide what is essential for the student :) A good way to go about this would be to check the example solution, as that should be solution you expect the student to write.